vamb/__main__.py

```diff
@@ -1442,7 +1447,41 @@ def run_bin_default(opt: BinDefaultOptions):
         FastaOutput.try_from_common(opt.common),
         None,
     )
-    del latent
+
+
+class RunDefault:
+    pass
+
+
+class RunTrain:
+    pass
+
+
+class RunCluster:
+    def __init__(self, latent: Path):
+        self.latent = latent
+
+
+def load_train_bin(
+    opt: BinDefaultOptions, partial_mode: Union[RunDefault, RunTrain, RunCluster]
+):
+    composition, abundance = load_composition_and_abundance(
+        vamb_options=opt.common.general,
+        comp_options=opt.common.comp,
+        abundance_options=opt.common.abundance,
+        binsplitter=opt.common.output.binsplitter,
+    )
+
+    latent = None
+
+    if isinstance(partial_mode, (RunDefault, RunTrain)):
+        latent = run_train_vae(opt, composition, abundance)
+
+    if isinstance(partial_mode, RunCluster):
+        latent = partial_mode.latent
```

Comment:
Things I would change here:
1. Delete the `latent = None` line. This is such that, the type checker will check that `latent` is not defined, unless we cover all possibilities with the if statements below.
2. Make sure `latent` is always an `np.ndarray`, and never a `Path`.

So, something like:
```python
if isinstance(partial_mode, (RunDefault, RunTrain)):
    latent = run_train_vae(opt, composition, abundance)

if isinstance(partial_mode, RunCluster):
    latent = vamb.vambtools.read_npz(partial_mode.latent_path)
```